src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -262,10 +262,10 @@ class Instruction_aarch64 {
     int nbits = msb - lsb + 1;
     int64_t chk = val >> (nbits - 1);
     guarantee (chk == -1 || chk == 0, "Field too big for insn");
-    unsigned uval = val;
+    uint64_t uval = val;
```

Comment:
```suggestion
    unsigned uval = checked_cast<unsigned>((uint64_t)val);
```

Reply:
uint64_t uval64 = val;
    unsigned uval = checked_cast<unsigned>(uval64);
This won't work for negative values, right?

Reply:
val comes in signed, so we want to just chop off the sign.  checked_cast<unsigned>(signed val) will assert.  checked_cast<> doesn't do sign conversion.  We don't have a cast that does sign conversion.

Reply:
Not sure I trust this either.  I'm going to write a gtest for this.

Reply:
We already have a havrd guarantee that this signed value fits in the signed field. There's no need for any more asserting for that: just chop it off.